Source/WebKit/Shared/WebGPU/WebGPUExternalTextureDescriptor.h

```diff
@@ -36,7 +36,7 @@
 namespace WebKit::WebGPU {
 
 struct ExternalTextureDescriptor : public ObjectDescriptorBase {
-    PAL::WebGPU::VideoSourceIdentifier mediaIdentifier;
+    uint64_t mediaIdentifier;
```

Comment:
So here is a definition of the external texture: obtain the texture from identifier.
.. except, if the message had sharedFrame, in that case obtain it from the shared frame.

This is wildly incoherent.

Note, your corresponding "ideal world model" is :

GPUExternalTextureDescriptor
* GPUVideoSource source;
* GPUPredefinedColorSpace colorSpace

So you should have corresponding:

WebKit::WebGPU::ExternalTextureDescriptor
* WebKit::WebGPU::VideoSource source
* PAL::WebGPU::PredefinedColorSpace colorSpace

using WebKit::WebGPU::VideoSource = std::variant<PAL::WebGPU::VideoSourceIdentifier, WebKit::SharedVideoFrame>;